### PR TITLE
Modularize resources for better re-use

### PR DESCRIPTION
**Pull request by chasinglogic**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -38,54 +38,52 @@ resource "aws_subnet" "this_public_subnet" {
...
-  vpc_peering_connection_id = "${aws_vpc_peering_connection.this_vpc_peer.id}"
-  auto_accept               = true
+  providers = {
+    "aws.primary" = "aws.primary"
+    "aws.peer"    = "aws.peer"
+  }
 
-  tags = "${merge(var.vpc_peer_connection_tags, var.tags, map("Side", "Accepter"), map("Name", format("%s.peer_to_%s", var.peer_vpc_id, var.name)))}"
+  name = "${var.name}"
+  primary_vpc_id = "${var.peer_vpc_id}"
```
Comment (zakhark) [inline main.tf]: This is confusing because on the next line there is also peer_vpc_id that's not ${var.peer_vpc_id}

File: main.tf [hunk 2]
```diff
@@ -38,54 +38,52 @@ resource "aws_subnet" "this_public_subnet" {
...
-  gateway_id                = "${aws_internet_gateway.this_inet_gw.id}"
-}
+  providers = {
+    "aws.primary" = "aws.primary"
+    "aws.peer"    = "aws.peer"
+  }
+
+  # The way the sub module and this parent one refer to "primary" and
+  # "peer" is reversed since the perspective is a little different.
+  primary_route_table_id            = "${var.peer_route_table}"
```
Comment (zakhark) [inline main.tf]: peer_route_table deviates from the naming convention used here - %some_resource%_id

**Review by zakhark — APPROVED** [review]
lgtm modulo comments

File: main.tf (hunk 1 above)
Comment (chasinglogic) [reply main.tf]: Added a clarifying comment like the one on Line #76

File: main.tf (hunk 2 above)
Comment (chasinglogic) [reply main.tf]: Renamed

**Review by briananthonymccarthy — APPROVED** [review]
lgtm

**Pull request merged**